autoload/pythonimports.vim

```diff
@@ -6,12 +6,35 @@ function! pythonimports#filename2module(filename)
   " directly included in PYTHONPATH.
   let pkg = fnamemodify(a:filename, ":p")
   let root = fnamemodify(pkg, ":h")
-  while strlen(root)
-    if !filereadable(root . "/__init__.py")
-      break
-    endif
-    let root = fnamemodify(root, ":h")
-  endwhile
+
+  let found = 0
+  if exists("g:python_paths")
+      for path in g:python_paths
+            if root =~ path
```

Comment:
This is a regex match?  I think it should be a prefix match.

Reply:
I see your inner loop that checks for `== path` ensures the match is a prefix, even if you got a false positive here.

Reply:
No, it's not a regex match. I wanted to check, if root path **contains** a path from python_paths. I didn't know about `=~` operator, just read about it from https://stackoverflow.com/a/9527672/3110300. You may chek it in vim:
```
:echo "/home/arthur/repos/folder" =~ "/home/arthur/repos" # gives 1
:echo "/home/arthur/repos/" =~ "/home/arthur/repos" # 1
:echo "/home/arthur/repos" =~ "/home/arthur/repos" # 1
:echo "/home/arthur/" =~ "/home/arthur/repos" # 0
```

Reply:
maybe there's a better way to check if A is a substring of B?